src/shared/fstab-util.c

```diff
@@ -34,6 +34,21 @@
 #include "strv.h"
 #include "util.h"
 
+bool fstab_has_fstype(const char *fstype) {
+        _cleanup_endmntent_ FILE *f = NULL;
+        struct mntent *m;
+
+        f = setmntent("/etc/fstab", "re");
+        if (!f)
+                return false;
+
+        while ((m = getmntent(f)))
```

Comment:
and this should maybe read:

```c
for (;;) {
        errno = 0;
        m = getmntent(f);
        if (!m)
                return errno != 0 ? -errno : false;
        …
``` 

The initialization of errno is necessary so that we can distuingish an error from EOF